models/models.go

```diff
@@ -10,58 +10,58 @@ import (
 
 // Filter represents a structure for a filter job
 type Filter struct {
-	InstanceID       string      `json:"instance_id"`
-	DimensionListURL string      `json:"dimension_list_url,omitempty"`
-	Dimensions       []Dimension `json:"dimensions,omitempty"`
-	Downloads        Downloads   `json:"downloads,omitempty"`
-	Events           Events      `json:"events,omitempty"`
-	FilterID         string      `json:"filter_job_id,omitempty"`
-	State            string      `json:"state,omitempty"`
-	Links            LinkMap     `json:"links,omitempty"`
+	InstanceID       string      `bson:"instance_id"          json:"instance_id"`
+	DimensionListURL string      `bson:"dimension_list_url"   json:"dimension_list_url,omitempty"`
+	Dimensions       []Dimension `bson:"dimensions,omitempty" json:"dimensions,omitempty"`
+	Downloads        Downloads   `bson:"downloads"            json:"downloads,omitempty"`
+	Events           Events      `bson:"events"               json:"events,omitempty"`
+	FilterID         string      `bson:"filter_job_id"        json:"filter_job_id,omitempty"`
+	State            string      `bson:"state"                json:"state,omitempty"`
+	Links            LinkMap     `bson:"links"                json:"links,omitempty"`
 }
 
 // LinkMap contains a named LinkObject for each link to other resources
 type LinkMap struct {
-	Version LinkObject `json:"version,omitempty"`
+	Version LinkObject `bson:"version" json:"version,omitempty"`
 }
 
 // LinkObject represents a generic structure for all links
 type LinkObject struct {
-	ID   string `json:"id,omitempty"`
-	HRef string `json:"href,omitempty"`
+	ID   string `bson:"id"   json:"id,omitempty"`
+	HRef string `bson:"href" json:"href,omitempty"`
 }
 
 // Dimension represents an object containing a list of dimension values and the dimension name
 type Dimension struct {
-	DimensionURL string   `json:"dimension_url,omitempty"`
-	Name         string   `json:"name,omitempty"`
-	Options      []string `json:"options,omitempty"`
+	DimensionURL string   `bson:"dimension_url"           json:"dimension_url"`
+	Name         string   `bson:"name"                    json:"name"`
+	Options      []string `bson:"options"                 json:"options"`
 }
 
 // Downloads represents a list of file types possible to download
 type Downloads struct {
```

Comment:
Again all Download types to be omitempty for both bson and json tags.

Reply:
Doing this will break one of the requirements in trello. See https://trello.com/c/ovo8dl13/1997-update-filter-api-for-multiple-output-types-p8. It states ` Add empty download objects for all available file types to job once user has submitted`

Reply:
Okay fair enough